### PR TITLE
Add include_text parameter to SFT dataloaders

### PR DESCRIPTION
**Pull request by Kipok**
# What does this PR do ?

Adding a new parameter to include original text in the dictionary returned from a dataloader. This is needed for a cleaner support of metrics in NeMo-Aligner.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ def __init__(
         truncation_method: str = 'right',
         special_tokens: Optional[Mapping[str, str]] = None,  # special tokens, a dictory of {token_type: token}
         is_test: bool = False,
+        include_text: bool = False,
```
Comment (cuichenx) [inline nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py:61]: can you rename this to "output_original_text" or something to make more clear

**Comment by gshennvm** [comment]
jenkins

**Review by cuichenx — CHANGES_REQUESTED** [review]
Can you rename ` include_text` to "output_original_text" or something similar to make more clear

File: nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py (hunk 1 above)
Comment (Kipok) [reply nemo/collections/nlp/data/language_modeling/megatron/gpt_sft_dataset.py:61]: sure!

**Comment by gshennvm** [comment]
jenkins

**Review by cuichenx — APPROVED** [review]
LGTM!

**Pull request merged**